dnsenum: add page (#9195)
tldr-pages/tldr: pages/linux/dnsenum.md

# dnsenum

> > Multithreaded perl script used to enumerate DNS information from a domain.

Reviewer: ```suggestion
> Multithreaded Perl script used to enumerate DNS information from a domain.
```

> More information: <https://github.com/SparrowOchon/dnsenum2>.

- Enumerate a domain with default settings:

> `dnsenum -enum {{domainname}`

Reviewer 2: I think an explicit example is better here
```suggestion
`dnsenum -enum {{example.com}`
```

> `dnsenum -enum {{example.com}`

Reviewer 2: typo
```suggestion
`dnsenum -enum {{example.com}}`
```

> - Enumerate a domain with default settings:
> 
> `dnsenum -enum {{example.com}}`

Reviewer 3: ```suggestion
- [enum]erate a specific domain with default settings:

`dnsenum -enum {{example.com}}`
```

Optional fix: mnemonic can be used.
Optional fix: `specific` may sound better.

> - Enumerate domain without reverse lookup and save the output to file:

Reviewer 2: ```suggestion
- Enumerate a domain without reverse lookup and save the output to file:
```

> `dnsenum --noreverse -o {{outputfile}} {{domainname}}`

Reviewer 2: ```suggestion
`dnsenum --noreverse -o {{path/to/file}} {{example.com}}`
```

> - Enumerate a domain without reverse lookup and save the output to a file:
> 
> `dnsenum --noreverse -o {{path/to/file}} {{example.com}}`

Reviewer 3: ```suggestion
- Enumerate a domain without reverse lookup and save the [o]utput to a specific file:

`dnsenum --noreverse -o {{path/to/file}} {{example.com}}`
```

Almost the same issue as above.
Warning: [parsing man page failed, can't find `-o` option, possibly broken manual](https://manned.org/dnsenum.1).

> - Brute force search on subdomains along with custom file with subdomains passed as attribute:

Reviewer: ```suggestion
- Brute force search on subdomains along with a custom file with subdomains passed as an attribute:
```

> `dnsenum -f {{filename}} -r {{domainname}}`

Reviewer 2: ```suggestion
`dnsenum -f {{filename}} -r {{example.com}}`
```

> - Brute force search on subdomains along with a custom file with subdomains passed as an attribute:
> 
> `dnsenum -f {{filename}} -r {{example.com}}`

Reviewer 3: ```suggestion
- Brute force search on subdomains along with a specific file with subdomains passed as an attribute:

`dnsenum --file {{path/to/file}} --recursion {{example.com}}`
```

Almost the same issue as above.
Mandatory fix: [path format is broken](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#paths).
Mandatory fix: [long option must be used if it's portable](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#option-syntax).
